src/XrdCms/XrdCmsCluster.cc

```diff
@@ -1816,12 +1833,29 @@ XrdCmsNode *XrdCmsCluster::SelbyLoad(SMask_t mask, XrdCmsSelector &selR)
                       }
                    Multi = true;
                   }
+               }
           }
-
+    if (Config.P_randlb==1){ 
+    // pick a random weighed node
+    //
+    std::random_device rand_dev;
+    std::mt19937 generator(rand_dev());
```

Comment:
This function must be called under a lock, so these can probably be created at initialization only and be reused afterwards. Also, I'd avoid hard-coding [`mt19937`](https://en.cppreference.com/w/cpp/header/random) here, which is 32-bit only, and just use `default_random_engine` instead. Frequent initialization with hardware-based randomness can create unnecessary slowness.

Reply:
switched the random number generator to static